app/(app)/parking-cards/verification/index.tsx

```diff
@@ -12,57 +13,77 @@ import Panel from '@/components/shared/Panel'
 import Typography from '@/components/shared/Typography'
 import { useTranslation } from '@/hooks/useTranslation'
 import { clientApi } from '@/modules/backend/client-api'
-import { VerifyEmailsDto } from '@/modules/backend/openapi-generated'
-
-/*
- * TODO
- * - [ ] add email validation
- * - [ ] handle error on mutation (email not sent)
- */
+import { SERVICEERROR, VerifyEmailsDto } from '@/modules/backend/openapi-generated'
+import { isServiceError } from '@/utils/errorService'
+import { isValidEmail } from '@/utils/isValidEmail'
 
 const Page = () => {
   const t = useTranslation('AddParkingCards')
 
   const [email, setEmail] = useState('')
 
-  // TODO add email validation
-  const isValid = email.includes('@')
+  const [expectedError, setExpectedError] = useState<string | null>(null)
 
-  // TODO deduplicate this mutation (it's also used in link-expired.tsx)
+  // TODO deduplicate this mutation (it's also used in verification-result.tsx)
   const mutation = useMutation({
     mutationFn: (bodyInner: VerifyEmailsDto) =>
       clientApi.verifiedEmailsControllerSendEmailVerificationEmails(bodyInner),
     onSuccess: (res) => {
       const tmpVerificationToken = res.data[0].token
 
-      router.replace({
+      router.push({
         pathname: '/parking-cards/verification/verification-sent',
         params: {
           email,
           tmpVerificationToken,
         },
       })
     },
+    onError: (error) => {
+      if (
+        isAxiosError(error) &&
+        isServiceError(error.response?.data) &&
+        error.response?.data.errorName === SERVICEERROR.EmailAlreadyVerified
+      )
+        setExpectedError('EmailAlreadyVerified')
+      else setExpectedError('GeneralError')
+    },
   })
 
-  const handleSendVerificationEmail = () => {
-    const body: VerifyEmailsDto = {
-      emails: [email],
+  const onChangeText = (val: string) => {
```

Comment:
We use naming as: `handle` for functions, `on` for props.
Please no abbreviations (val --> value)